Ensure awaysite overmap grids are clear of hazards.

MuckerMayhem opened the pull request:
🆑 Mucker
tweak: Visitable away sites will no longer sometimes spawn with hazards over top them.
/🆑

Deletes overmap hazards on the same grid as a visitable away site, if the site spawns on one. Doesn't prevent them from being surrounded, but it should allow access to sites more often for more parties, without negating the need of the OFD from time to time.

Spookerton commented on code/modules/overmap/sectors.dm:
```diff
@@ -40,6 +40,9 @@
 
 	forceMove(locate(start_x, start_y, GLOB.using_map.overmap_z))
 
+	for(var/obj/effect/overmap/event/E in locate(src.x, src.y, src.z))
```
```suggestion
	for(var/obj/effect/overmap/event/E in loc)
```

Spookerton commented:
Not sure if want, though.

MuckerMayhem commented:
> Not sure if want, though.

Goal is to mainly allow non-torch ships more consistent access to planets/sites that aren't surrounded by hazards, and maybe are just on the fringe. Many GUP users will tell you it sucks immensely to get to a target after a long drive just to find it under a space dust field.

Another pitched idea was to make it so this only affects space dust and electrical storms, as all other hazards can be maneuvered around.

MuckerMayhem committed:
Ensure awaysite overmap grids are clear of hazards.

Deletes overmap hazards if an away site spawns on one. Doesn't prevent them from being surrounded, but it should allow more access more of the time without negating the need for the OFD.

Spookerton reviewed (approved):
bweh.
